core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -81,25 +126,87 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
         if (negativeQuery == null) {
```

Comment:
agreed, sorry I keep getting confused about this. those queries can be null, we do these checks to prevent NPEs, makes sense.